File: verify_errors.py
from playwright.sync_api import sync_playwright
import time

def verify_models():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        # Listen for console logs and errors
        page.on("console", lambda msg: print(f"PAGE LOG: {msg.text}"))
        page.on("pageerror", lambda err: print(f"PAGE ERROR: {err.message}"))

        try:
            print("Navigating to test harness...")
            page.goto("http://127.0.0.1:8000/docs/test_models.html")

            # Wait for loading overlay to disappear
            page.wait_for_selector("#loading-overlay", state="hidden", timeout=10000)

            print("Selecting Genetic Model...")
            page.select_option("#model-selector", "genetic")

            # Wait for some time for the model to load and start
            time.sleep(5)
            page.screenshot(path="/home/jules/genetic_model_check.png")

            print("Selecting Neuro Model...")
            page.select_option("#model-selector", "neuro")

            time.sleep(5)
            page.screenshot(path="/home/jules/neuro_model_check.png")

        except Exception as e:
            print(f"Error during verification: {e}")
        finally:
            browser.close()

if __name__ == "__main__":
    verify_models()

File: verify_errors.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  // Listen for console logs and errors
  page.on('console', msg => console.log('PAGE LOG:', msg.text()));
  page.on('pageerror', err => console.log('PAGE ERROR:', err.message));

  await page.goto('http://127.0.0.1:8000/docs/test_models.html');

  // Wait for loading overlay to disappear
  await page.waitForSelector('#loading-overlay', { state: 'hidden', timeout: 10000 });

  console.log('Selecting Genetic Model...');
  await page.selectOption('#model-selector', 'genetic');

  // Wait for some time for the model to load and start
  await page.waitForTimeout(5000);

  await page.screenshot({ path: '/home/jules/genetic_model_check.png' });

  console.log('Selecting Neuro Model...');
  await page.selectOption('#model-selector', 'neuro');

  await page.waitForTimeout(5000);

  await page.screenshot({ path: '/home/jules/neuro_model_check.png' });

  await browser.close();
})();
